{"kind":"symbol","abstract":[{"type":"text","text":"Completes computation exceptionally with a specified exception."},{"text":" ","type":"text"},{"type":"text","text":"This method does nothing if computation has already "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","isActive":true,"type":"reference"},{"type":"text","text":" completed,"},{"type":"text","text":" "},{"text":"otherwise ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()"},{"text":" will return the specified exception.","type":"text"}],"schemaVersion":{"major":0,"patch":0,"minor":3},"identifier":{"url":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","interfaceLanguage":"swift"},"hierarchy":{"paths":[["doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise"]]},"metadata":{"roleHeading":"Instance Method","role":"symbol","externalID":"s:15DXFeedFramework7PromiseC21completeExceptionallyyyAA14GraalExceptionOKF","symbolKind":"method","title":"completeExceptionally(_:)","modules":[{"name":"DXFeedFramework"}],"fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"text":"completeExceptionally","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException"},{"kind":"text","text":") "},{"kind":"keyword","text":"throws"}]},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"completeExceptionally"},{"text":"(","kind":"text"},{"text":"_","kind":"externalParam"},{"text":" ","kind":"text"},{"kind":"internalParam","text":"exception"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException","text":"GraalException","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO"},{"text":") ","kind":"text"},{"kind":"keyword","text":"throws"}],"platforms":["macOS"],"languages":["swift"]}]},{"kind":"content","content":[{"text":"Discussion","type":"heading","anchor":"discussion","level":2},{"type":"unorderedList","items":[{"content":[{"type":"paragraph","inlineContent":[{"text":"Throws : GraalException. Rethrows exception from Java","type":"text"}]}]}]}]}],"variants":[{"paths":["\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"references":{"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/complete(result:)":{"role":"symbol","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","type":"topic","abstract":[{"type":"text","text":"Completes computation normally with a specified result."},{"text":" ","type":"text"},{"type":"text","text":"This method does nothing if computation has already "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","type":"reference","isActive":true},{"text":" completed","type":"text"},{"type":"text","text":" "},{"type":"text","text":"(normally, exceptionally, or was cancelled),"}],"url":"\/documentation\/dxfeedframework\/promise\/complete(result:)","title":"complete(result:)","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"complete","kind":"identifier"},{"kind":"text","text":"("},{"text":"result","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:15DXFeedFramework11MarketEventC","text":"MarketEvent","kind":"typeIdentifier"},{"kind":"text","text":") "},{"text":"throws","kind":"keyword"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/cancel()":{"role":"symbol","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","type":"topic","url":"\/documentation\/dxfeedframework\/promise\/cancel()","abstract":[{"type":"text","text":"Cancels computation. This method does nothing if computation has already "},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","isActive":true},{"type":"text","text":" completed."}],"title":"cancel()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"cancel"},{"text":"()","kind":"text"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/hasException()":{"fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"hasException","kind":"identifier"},{"text":"() -> ","kind":"text"},{"preciseIdentifier":"s:Sb","text":"Bool","kind":"typeIdentifier"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"Returns "},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"type":"text","text":" when computation has completed exceptionally or was cancelled."},{"text":" ","type":"text"},{"text":"Use ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","type":"reference","isActive":true},{"text":" method to get the exceptional outcome of the computation.","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()","url":"\/documentation\/dxfeedframework\/promise\/hasexception()","role":"symbol","title":"hasException()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/GraalException":{"fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"GraalException","kind":"identifier"}],"role":"symbol","abstract":[{"type":"text","text":"Represents errors that occur when calling graal function."}],"type":"topic","url":"\/documentation\/dxfeedframework\/graalexception","kind":"symbol","title":"GraalException","navigatorTitle":[{"kind":"identifier","text":"GraalException"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isCancelled()":{"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"isCancelled","kind":"identifier"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}],"kind":"symbol","type":"topic","abstract":[{"text":"Returns ","type":"text"},{"type":"strong","inlineContent":[{"text":"true","type":"text"}]},{"text":" when computation was canceled.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"Use "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","type":"reference","isActive":true},{"type":"text","text":" method to get the corresponding exception."}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()","url":"\/documentation\/dxfeedframework\/promise\/iscancelled()","role":"symbol","title":"isCancelled()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/completeExceptionally(_:)":{"role":"symbol","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","type":"topic","abstract":[{"text":"Completes computation exceptionally with a specified exception.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"This method does nothing if computation has already "},{"isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","type":"reference"},{"type":"text","text":" completed,"},{"type":"text","text":" "},{"type":"text","text":"otherwise "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()"},{"type":"text","text":" will return the specified exception."}],"url":"\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)","title":"completeExceptionally(_:)","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"completeExceptionally"},{"text":"(","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException"},{"kind":"text","text":") "},{"text":"throws","kind":"keyword"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise":{"fragments":[{"kind":"keyword","text":"class"},{"text":" ","kind":"text"},{"text":"Promise","kind":"identifier"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"Result of a computation that will be completed normally or exceptionally in the future."},{"text":" ","type":"text"},{"text":"This class is designed to represent a promise to deliver certain result.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"Promise"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise","url":"\/documentation\/dxfeedframework\/promise","role":"symbol","title":"Promise"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/getException()":{"fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"getException","kind":"identifier"},{"kind":"text","text":"() -> "},{"preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"Returns exceptional outcome of computation. If computation has no  "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()"},{"text":" exception,","type":"text"},{"type":"text","text":" "},{"text":"then this method returns nil. If computation has completed exceptionally or was cancelled, then","type":"text"},{"type":"text","text":" "},{"type":"text","text":"the result of this method is not nil."},{"type":"text","text":" "},{"type":"text","text":"If computation was "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()"},{"text":" cancelled, then this method returns an","type":"text"},{"text":" ","type":"text"},{"text":"instance of GraalException.","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","url":"\/documentation\/dxfeedframework\/promise\/getexception()","role":"symbol","title":"getException()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isDone()":{"role":"symbol","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","type":"topic","url":"\/documentation\/dxfeedframework\/promise\/isdone()","abstract":[{"type":"text","text":"Returns "},{"inlineContent":[{"text":"true","type":"text"}],"type":"strong"},{"text":" when computation has","type":"text"},{"text":" ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","isActive":true,"type":"reference"},{"type":"text","text":" completed normally,"},{"type":"text","text":" "},{"text":"or ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)"},{"type":"text","text":" exceptionally,"},{"type":"text","text":" "},{"text":"or was ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"type":"text","text":" canceled"}],"title":"isDone()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"isDone"},{"kind":"text","text":"() -> "},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework":{"role":"collection","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","abstract":[],"url":"\/documentation\/dxfeedframework","type":"topic","title":"DXFeedFramework","kind":"symbol"}}}